### PR TITLE
Run3-alca248 Attach a parameter to write the content in the tree for tracks only in a given momentum range

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Attach a parameter to write the content in the tree for tracks only in a given momentum range

#### PR validation:

Tested with the latest AlCaReco files

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 1]
```diff
@@ -256,9 +259,15 @@ void HcalIsoTrackAnalyzer::analyze(edm::Event const& iEvent, edm::EventSetup con
       if (debug)
         edm::LogVerbatim("HcalIsoTrack") << "eHcal:eHcal10:eHCal30 " << t_eHcal << ":" << t_eHcal10 << t_eHcal30;
 #endif
-      tree->Fill();
-      edm::LogVerbatim("HcalIsoTrackX") << "Run " << t_Run << " Event " << t_Event << " p " << t_p;
-
+      bool select(true);
+      if (fillInRange_) {
+        if ((t_p < pTrackLow_) || (t_p < pTrackHigh_))
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:264]:
```suggestion
        if ((t_p < pTrackLow_) || (t_p > pTrackHigh_))
```

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 2]
```diff
@@ -256,9 +259,15 @@ void HcalIsoTrackAnalyzer::analyze(edm::Event const& iEvent, edm::EventSetup con
...
+          select = false;
+      }
+      if (select) {
+        tree->Fill();
+        edm::LogVerbatim("HcalIsoTrackX") << "Run " << t_Run << " Event " << t_Event << " p " << t_p;
+      }
       if (t_p < pTrackLow_) {
         ++nLow_;
       } else if (t_p < pTrackHigh_) {
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:273]:
```suggestion
      } else if (t_p > pTrackHigh_) {
```

**Review by perrotta — COMMENTED** [review]
I think a fix is needed (also in the code previous to this PR):  please check!

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc]: Thanks - taking care for it

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
+alca

**Comment by rappoccio** [comment]
+1

**Pull request merged**